### DISCUSSION TITLE
Use of null propagation and out variables

### DISCUSSION THREAD

**Discussion by israellot** [dotnet/csharplang#4214] [Q&A] [answered]
Couple of thing I would like to discuss and understand

Here a small sample of using out variable declaration just for base 
``` 
var dictionary = new Dictionary<string, string>();
if(dictionary.TryGetValue("key",out var s))
{
 Console.Write(s); 
}
```

Now if for some reason I expect *dictionary* to be null and I would use null propagation 
``` 
var dictionary = new Dictionary<string, string>();
if(dictionary?.TryGetValue("key",out var s)==true)
{
 Console.Write(s); 
}
```

First thing that bogs me is that I have to explicitly compare to true since now the result is nullable<bool>. 
Second thing is that now s is undefined inside the *if*, and I would have to write it as 
``` 
var dictionary = new Dictionary<string, string>();
string s =null;
if(dictionary?.TryGetValue("key",out s)==true)
{
 Console.Write(s); 
}
```

Am I missing something or null propagation and out variables were never meant to play well together?

**Comment by 333fred** [comment, marked as answer]
@RikkiGibson is actually working on a proposal to make this better. There are number of related issues on csharplang in this area.

**Reply by RikkiGibson** [reply]
The proposal is now available at https://github.com/dotnet/csharplang/discussions/4240.
